hazelcast/src/main/java/com/hazelcast/map/impl/NearCache.java

```diff
@@ -53,12 +57,10 @@
     private static final int HUNDRED_PERCENT = 100;
     private static final int EVICTION_PERCENTAGE = 20;
     private static final int CLEANUP_INTERVAL = 5000;
-    private final int maxSize;
     private volatile long lastCleanup;
-    private final long maxIdleMillis;
-    private final long timeToLiveMillis;
     private final EvictionPolicy evictionPolicy;
     private final InMemoryFormat inMemoryFormat;
+    private final MapContainer mapContainer;
```

Comment:
That's correct. MapContainer need to be kept because NearCacheConfig will get invalide/out of sync in case if new mapconfig was set.